packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -256,7 +256,6 @@ class GameWidgetState<T extends Game> extends State<GameWidget<T>> {
 
   void disposeCurrentGame() {
     currentGame.removeGameStateListener(_onGameStateChange);
-    currentGame.onRemove();
```

Comment:
This was the one you thought never could be called right?

Reply:
Yes, that is not showing in my file.  Let me see why the file diff is not being picked up.

Reply:
It is showing as removed in GitHub too, just checking :)

Reply:
> This was the one you thought never could be called right?

I also just want to add for others reading this, it does get called for many situations, but the bug was, it does NOT get called when a route is destroyed that contains the game.